benchmarks/benchmarks/bench_centrality.py

```diff
@@ -0,0 +1,22 @@
+import networkx as nx
+import nx_parallel as nxp
+
+
+class BetweennessCentralityBenchmark:
+    params = [("parallel", "normal")]
+    param_names = ["algo_type"]
+
+    def setup(self, algo_type):
+        self.algo_type = algo_type
+
+    def time_betweenness_centrality(self, algo_type):
+        num_nodes, edge_prob = 300, 0.5
+        G = nx.fast_gnp_random_graph(num_nodes, edge_prob, directed=False)
+        if algo_type == "parallel":
```

Comment:
Maybe we should also have a third algo_type, where we run it using a keyword instead of converting the graph to a parallel_graph.  That is, the call would be something like:

```python
        - = nx.betweenness_centrality(G, backend=‘nx-parallel’)
```
I guess we better make sure that works first, and gives the same results. But it should work and it’d be nice to know if they are the same speed.  Of course, if they are almost identical all the time, then we probably should remove it again.

Reply:
@dschult 
the output is same for both the ways(i.e. converting into a parallel graph and using keyword argument)when I round the betweenness_centrality value up to 2 decimal places, otherwise, it does give different values sometimes. 

and yes, both the ways are almost identical in terms of time, but I've still added it in the comments [here](https://github.com/networkx/nx-parallel/pull/24/commits/7857b4a460f6ae3ba3409767acd34de730f99a7d#diff-a7bf9379a7c75e76deb05248bd8e6e018df136139dcf8281df36b1f8de32bfafR25) if someone wants to try it. There is some difference for 500 nodes graphs as observed in the following ss :
<img width="1438" alt="Screenshot 2023-11-03 at 5 24 33 PM" src="https://github.com/networkx/nx-parallel/assets/91629733/1098fe9b-3830-4ea2-bd44-02c0787ff3ea">

But this difference seem negligible when we bring sequential into the picture(as seen in the following ss) : 
<img width="1438" alt="Screenshot 2023-11-03 at 5 26 34 PM" src="https://github.com/networkx/nx-parallel/assets/91629733/a6393d20-ae36-4317-a979-868aa337c6bd">

also, i have added num_nodes and edge_prob and changed `normal` algo_type to `sequential` in this [new commit](https://github.com/networkx/nx-parallel/pull/24/commits/7857b4a460f6ae3ba3409767acd34de730f99a7d)

Thank you :)

Reply:
also, should I create a PR to change the plugin name from `parallel` to `nx-parallel`, if that's required?

Reply:
Thanks for those time comparisons.  I'm glad it doesn't change the time much between using the backend by object type and by using it via a keyword arg.

I think the naming conventions for these backend libraries is supposed to be, e.g.  `nx-cugraph` when the plugin name is `cugraph`. I will check with other people though.

The additional parameter `graph_type`s you mention are already obtained from the graph object. So the backend gets that info via, e.g. `G.is_directed()`.  

Graph generators will need to be done either in networkx before calling the function, or in nx-parallel by writing a parallel version of that function. I believe the PR with the graph generators getting @dispatch decorators has been merged.  The weight parameter should be passed through to the backend function. If it is None then we use an unweighted approach.  

I haven't looked at the new commit yet, but hope to soon. :}